Integrations/Forescout/Forescout.py

```diff
@@ -0,0 +1,678 @@
+
+    Parameters
+    ----------
+    method : str
+        HTTP method, e.g. 'GET', 'POST' ... etc.
+    url_suffix : str
+        API endpoint.
+    full_url : str
+        Bypasses the use of BASE_URL + url_suffix. Useful if there is a need to
+        make a request to an address outside of the scope of the integration
+        API.
+    headers : dict
+        Headers to send in the request.
+    auth : tuple
+        Auth tuple to enable Basic/Digest/Custom HTTP Auth.
+    params : dict
+        URL parameters.
+    data : dict
+        Data to be sent in a 'POST' request.
+    files : dict
+        File data to be sent in a 'POST' request.
+    resp_type : str
+        Determines what to return from having made the HTTP request. The default
+        is 'json'. Other options are 'text', 'content' or 'response' if the user
+        would like the full response object returned.
+
+    Returns
+    -------
+    dict | str | bytes | obj
+        Response JSON from having made the request.
+    """
+    try:
+        address = full_url if full_url else BASE_URL + url_suffix
+        res = requests.request(
+            method,
+            address,
+            verify=USE_SSL,
+            params=params,
+            data=data,
+            files=files,
+            headers=headers,
+            auth=auth
+        )
+
+        # Handle error responses gracefully
+        if res.status_code not in {200, 201, 304}:
+            err_msg = 'Error in Forescout Integration API call [{}] - {}'.format(res.status_code, res.reason)
+            try:
+                # Try to parse json error response
+                res_json = res.json()
+                err_msg += '\n{}'.format(res_json.get('message'))
+                return_error(err_msg)
+            except json.decoder.JSONDecodeError:
+                if res.status_code in {400, 401, 501}:
+                    # Try to parse xml error response
+                    resp_xml = ET_PHONE_HOME.fromstring(res.content)
+                    demisto.info(str(res.content))
+                    codes = [child.text for child in resp_xml.iter() if child.tag == 'CODE']
+                    messages = [child.text for child in resp_xml.iter() if child.tag == 'MESSAGE']
+                    err_msg += ''.join([f'\n{code}: {msg}' for code, msg in zip(codes, messages)])
+                return_error(err_msg)
+
+        resp_type = resp_type.casefold()
+        if resp_type == 'json':
+            return res.json()
+        elif resp_type == 'text':
+            return res.text
+        elif resp_type == 'content':
+            return res.content
+        else:
+            return res
+
+    except requests.exceptions.ConnectionError:
+        err_msg = 'Connection Error - Check that the Server URL parameter is correct.'
+        return_error(err_msg)
+
+
+''' COMMANDS + REQUESTS FUNCTIONS '''
+
+
+def test_module():
+    """
+    Performs basic get request that requires proper authentication
+    """
+    login()
```

Comment:
i prefer we do some kind of get query here

Reply:
yea it's problematic for another reason which is that it only is testing the `Web API` essentially where in reality should be performing some kind of request to `DEX` as well - the problem with that is all the `DEX` API calls are to update stuff, not just retrieve some data...